### PR TITLE
Homework for Lecture 8 (SauceDemo Page Object)

### PR DESCRIPTION
**Pull request by kirill-zavelev**

### CODE REVIEW

File: src/test/java/com/saucedemo/test/CartTest.java [hunk 1]
```diff
@@ -14,46 +14,60 @@ public class CartTest extends BaseTest {
...
-        ProductPage productPage = new ProductPage(getDriver());
-        List<String> products = List.of("Sauce Labs Bike Light", "Sauce Labs Bolt T-Shirt", "Sauce Labs Fleece Jacket");
-        List<String> expectedProductPrices = productPage.getProductsPrices(products);
-        productPage.addProductsToCart(products);
-        CartPage cartPage = new CartPage(getDriver());
-        Assert.assertEquals(cartPage.getProductsPrices(products), expectedProductPrices,
-                "Prices of the products are not equal");
-        Assert.assertEquals(cartPage.getAllProductNames(), products,
-                "Names of the products are not equal");
+        final List<String> products = List.of("Sauce Labs Bike Light", "Sauce Labs Bolt T-Shirt",
```
Comment (testershmester) [inline src/test/java/com/saucedemo/test/CartTest.java]: static constants can be created for product names

File: src/test/java/com/saucedemo/test/CartTest.java [hunk 2]
```diff
@@ -14,46 +14,60 @@ public class CartTest extends BaseTest {
...
+                "Sauce Labs Fleece Jacket");
+        List<String> expectedProductsPrices = new LoginPage(getDriver())
+                .loginAsStandardUser()
+                .getProductsPrices(products);
+        CartPage cartPage = new ProductsPage(getDriver())
+                .addProductsToCart(products)
+                .openProductCart();
+        Assertions.assertThat(cartPage.getProductsPrices(products))
+                .isEqualTo(expectedProductsPrices)
+                .as("Products in cart should have the following prices: " + expectedProductsPrices);
```
Comment (testershmester) [inline src/test/java/com/saucedemo/test/CartTest.java]: move as before isEqual

Comment (testershmester) [reply src/test/java/com/saucedemo/test/CartTest.java]: image

File: src/test/java/com/saucedemo/test/CheckOutTest.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        List<String> expectedProductsNames = cartPage.getAllProductsNames();
+        cartPage.clickCheckOut()
+                .fillCheckoutForm("Name", "LastName", "12345")
+                .clickContinue();
+        CheckOutOverviewPage checkOutOverviewPage = new CheckOutOverviewPage(getDriver());
+        List<String> actualProductsNames = checkOutOverviewPage.getAllProductsNames();
+        List<String> actualProductsPrices = checkOutOverviewPage.getAllProductsPrices();
+        Assertions.assertThat(actualProductsPrices)
+                .isEqualTo(expectedProductsPrices)
+                .as("Prices should be " + expectedProductsPrices);
```
Comment (testershmester) [inline src/test/java/com/saucedemo/test/CheckOutTest.java]: image

File: src/main/java/com/saucedemo/page/CheckOutYourInformationPage.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        driver.findElement(POSTAL_CODE).sendKeys(postalCode);
+        return this;
+    }
+
+    public CartPage clickCancel() {
+        driver.findElement(CANCEL_BTN).click();
+        return new CartPage(driver);
+    }
+
+    public void clickContinue() {
```
Comment (testershmester) [inline src/main/java/com/saucedemo/page/CheckOutYourInformationPage.java]: what page is opened after clicking contnue?

**Review by testershmester — APPROVED** [review]
Excellent!

File: src/main/java/com/saucedemo/page/ProductsPage.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        return driver.findElement(TITLE_ELEMENT).getText();
+    }
+
+    public LeftSideMenu expandLeftSideMenu() {
+        driver.findElement(EXPAND_LEFT_SIDE_MENU).click();
+        return new LeftSideMenu(driver);
+    }
+
+    public List<String> sortProductsNamesAndGetList(String sortOption) {
+        switch (sortOption) {
```
Comment (testershmester) [inline src/main/java/com/saucedemo/page/ProductsPage.java:108]: Selenium class Select can be used

File: src/test/java/com/saucedemo/test/ProductsTest.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                .clickAddToCartForAllProducts();
+        List<WebElement> allProducts = productsPage.getAllProducts();
+        Assertions.assertThat(productsPage.getAmountOfAddedProductsDisplayedInCartIcon())
+                .isEqualTo(allProducts.size())
+                .as("Icon should display value " + allProducts.size());
+    }
+
+    @Test
+    public void checkProductsSorting() {
+        List<String> expectedProducts = new ArrayList<>();
```
Comment (testershmester) [inline src/test/java/com/saucedemo/test/ProductsTest.java]: constant is test can be added as initial list of all product names

File: src/test/java/com/saucedemo/test/ProductsTest.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        expectedProducts.add("Sauce Labs Bolt T-Shirt");
+        expectedProducts.add("Sauce Labs Fleece Jacket");
+        expectedProducts.add("Sauce Labs Onesie");
+        expectedProducts.add("Test.allTheThings() T-Shirt (Red)");
+        Collections.reverse(expectedProducts);
+        List<String> actualSortedProducts = new LoginPage(getDriver())
+                .loginAsStandardUser()
+                .sortProductsNamesAndGetList("Z - A");
+        Assertions.assertThat(actualSortedProducts)
+                .isEqualTo(expectedProducts)
```
Comment (testershmester) [inline src/test/java/com/saucedemo/test/ProductsTest.java]: check method isSorted in assertJ

**Review by testershmester — APPROVED** [review]

File: src/main/java/com/saucedemo/page/CheckOutYourInformationPage.java (hunk 4 above)
Comment (kirill-zavelev) [reply src/main/java/com/saucedemo/page/CheckOutYourInformationPage.java]: @testershmester indeed, CheckOutOverviewPage is opened. Thank you, will be fixed in next commit

**Pull request merged**